contexts/PaginationContext.tsx

```diff
@@ -0,0 +1,82 @@
+import { createContext, ReactNode, useState } from "react";
+import { PageStatesT, PaginateForT } from "types";
+
+export interface PaginationContextState {
+  pageStates: PageStatesT;
+  goToPage: (paginateFor: PaginateForT, page: number) => void;
+  nextPage: (paginateFor: PaginateForT) => void;
+  previousPage: (paginateFor: PaginateForT) => void;
+  setResultsPerPage: (paginateFor: PaginateForT, resultsPerPage: number) => void;
+}
+
+export const defaultPageState = {
+  number: 1,
```

Comment:
each piece of pageable data has a page number and a number of results per page. we use the formula `resultsPerPage * (pageNumber - 1)` in queries to determine what to fetch